portal-impl/src/com/liferay/portal/webserver/WebServerServlet.java

```diff
@@ -1482,6 +1436,40 @@ private static boolean _isDirectoryIndexingEnabled(Group group) {
 			PropsValues.WEB_SERVER_SERVLET_DIRECTORY_INDEXING_ENABLED);
 	}
 
+	private void _checkExpiredFileEntry(
```

Comment:
This is not a blocker for this pull, but more of an style issue for future prs (I'm ok with forwarding this as is). In general, we should try to avoid deep nesting like this. It is better to keep code as flat as possible to make is more sequential, making reading it easier; in this case, inverting the conditions would remove all unnecessary nesting:
```
		if (fileEntry == null) {
			return;
		}

		FileVersion fileVersion = fileEntry.getFileVersion();

		if (!fileVersion.isExpired()) {
			return;
		}

		User user = _getUser(httpServletRequest);

		PermissionChecker permissionChecker = _getPermissionChecker(
			httpServletRequest);

		if (permissionChecker.isContentReviewer(
				user.getCompanyId(), fileVersion.getGroupId()) ||
			Objects.equals(
				fileVersion.getUserId(), user.getUserId())) {

			return;
		}

		if (_log.isDebugEnabled()) {
			_log.debug(
				StringBundler.concat(
					"The file entry ", fileEntry.getFileEntryId(),
					" is expired, only users with content review ",
					"permissions can access to it."));
		}

		throw new FileEntryExpiredException(
			"The file entry " + fileEntry.getFileEntryId() +
				" is expired and user has no review permission");
```